tests/MainTest.elm

```diff
@@ -65,68 +50,236 @@ suite =
+        --             |> Expect.equal (Ok choice)
+        , Test.fuzz pairOfFormFieldFuzzer "encode old FormField can be decoded with decodeFormField" <|
+            \{ oldField, newField } ->
+                oldField
+                    |> encodeFormField
                     |> Json.Encode.encode 0
-                    |> Json.Decode.decodeString decodeChoice
-                    |> Expect.equal (Ok choice)
+                    |> Json.Decode.decodeString Main.decodeFormField
+                    |> Expect.equal (Ok newField)
         ]
 
 
 
 --
 
 
-viewModeFuzzer : Fuzzer ViewMode
+viewModeFuzzer : Fuzzer Main.ViewMode
 viewModeFuzzer =
     Fuzz.oneOf
         [ -- Fuzz.constant (Editor { maybeAnimate = Nothing })
           -- we don't encode/decode `maybeHighlight` because it is transient value
           -- maybeHighlight is always Nothing
-          Fuzz.constant (Editor { maybeAnimate = Nothing })
-        , Fuzz.constant Preview
-        , Fuzz.constant CollectData
+          Fuzz.constant (Main.Editor { maybeAnimate = Nothing })
+        , Fuzz.constant Main.Preview
+        , Fuzz.constant Main.CollectData
         ]
 
 
-inputFieldFuzzer : Fuzzer InputField
+inputFieldFuzzer : Fuzzer Main.InputField
 inputFieldFuzzer =
-    allInputField
+    Main.allInputField
         |> List.map Fuzz.constant
         |> Fuzz.oneOf
 
 
-presenceFuzzer : Fuzzer Presence
+presenceFuzzer : Fuzzer Main.Presence
 presenceFuzzer =
     Fuzz.oneOf
-        [ Fuzz.constant Required
-        , Fuzz.constant Optional
-        , Fuzz.map2 (\name description -> SystemRequired { name = name, description = description })
-            Fuzz.string
-            Fuzz.string
-        , Fuzz.map2 (\name description -> SystemOptional { name = name, description = description })
-            Fuzz.string
-            Fuzz.string
+        [ Fuzz.constant Main.Required
+        , Fuzz.constant Main.Optional
+        , Fuzz.constant Main.System
         ]
 
 
-fuzzFormField : Fuzzer FormField
+fuzzFormField : Fuzzer Main.FormField
 fuzzFormField =
-    Fuzz.map4 FormField
+    Fuzz.map5 Main.FormField
         string
+        (Fuzz.maybe string)
         presenceFuzzer
         string
         inputFieldFuzzer
 
 
-choiceStringFuzzer : Fuzzer Choice
+choiceStringFuzzer : Fuzzer Main.Choice
 choiceStringFuzzer =
     Fuzz.oneOf
-        [ Fuzz.map2 Choice
+        [ Fuzz.map2 Main.Choice
             Fuzz.string
             Fuzz.string
         , Fuzz.string
-            |> Fuzz.map (\s -> Choice s s)
+            |> Fuzz.map (\s -> Main.Choice s s)
+        ]
+
+
+
+-- Backward Compatible Types
+
+
+pairOfFormFieldFuzzer : Fuzzer { oldField : FormField, newField : Main.FormField }
+pairOfFormFieldFuzzer =
+    oldFormFieldFuzzer
+        |> Fuzz.map
+            (\oldField ->
+                { oldField = oldField
+                , newField = newFieldFromOldField oldField
+                }
+            )
```

Comment:
now we can create fuzzed values of old form fields, and also have an expected new form field to test assert against